cmd/cluster-autoscaler/cluster_autoscaler_test.go

```diff
@@ -50,18 +39,74 @@ func TestClusterAutoscalerScaleUpFromNonZero(t *testing.T) {
 
 	kc := h.KubernetesCluster(t)
 
+	kctl := testkit.NewKubectl(kc.KubeconfigPath)
+
 	k := testkit.NewKubernetes(kc.KubeconfigPath)
 	testkit.PollUntil(t, func() bool {
-		return len(k.ListReadyNodeNames(t)) > 1
+		// Only the control-plane node should be counted as Ready
+		return len(k.ListReadyNodeNames(t)) == 1
 	}, 20*time.Second)
 
 	helm := testkit.NewHelm(kc.KubeconfigPath)
 	// See https://github.com/kubernetes/autoscaler/tree/master/charts/cluster-autoscaler#tldr
+
+	helmInstallKwok(t, helm)
+	helmInstallClusterAutoscaler(t, helm, kctl)
+
+	os.Setenv("RESOURCE_NAME", appName)
+	os.Setenv("KUBECONFIG", kc.KubeconfigPath)
+	os.Setenv("NODE_LABEL_KEY", nodeLabelKey)
+	os.Setenv("NODE_LABEL_VALUE", nodeLabelValue)
+
+	logger := func() *logrus.Entry {
+		return logrus.NewEntry(logrus.New())
+	}
+	chatwork := &notify.Chatwork{
+		Logger: logger,
+	}
+	checker := cmd.NewChecker(context.Background(), false, logger, chatwork, "test", 7*time.Minute)
+	clusterautoscaler, err := NewClusterAutoscaler(checker)
+	if err != nil {
+		t.Fatalf("NewClusterAutoscaler: %s", err)
+	}
+
+	// Configure tolerations for kwok provider nodes
+	// This allows test pods to be scheduled on kwok-simulated nodes with taints
+	clusterautoscaler.SetDeploymentOption(DeploymentOption{
+		Tolerations: []apiv1.Toleration{
+			{
+				Key:      "kwok-provider",
+				Operator: apiv1.TolerationOpEqual,
+				Value:    "true",
+				Effect:   apiv1.TaintEffectNoSchedule,
+			},
+		},
+	})
```

Comment:
Nit, but can we do this instead, so that we don't need `SetDeploymentOption` for now?

```suggestion
	clusterautoscaler.DeploymentOption.Tolerations = []apiv1.Toleration{
		{
			Key:      "kwok-provider",
			Operator: apiv1.TolerationOpEqual,
			Value:    "true",
			Effect:   apiv1.TaintEffectNoSchedule,
		},
	}
```